ush/python/pygfs/task/analysis.py

```
@@ -328,3 +356,31 @@ def tgz_diags(statfile: str, diagdir: str) -> None:
             # Add diag files to tarball
             for diagfile in diags:
                 tgz.add(diagfile, arcname=os.path.basename(diagfile))
+
+
+@logit(logger)
+def find_value_in_nested_dict(nested_dict: Dict, key: str) -> Any:
+    """
+    Recursively search through a nested dictionary and return the value for the target key.
+    Parameters
+    ----------
+    nested_dict : Dict
+        Dictionary to search
+    key : str
+        Key to search for
+
+    Returns
+    -------
+    Any
+        Value of the key
+
+    Raises
+    ------
+    KeyError
+        If key is not found in dictionary
+
+    TODO: move this to a utility module so it can be used elsewhere
+    """
+    if not isinstance(nested_dict, dict):
+        return None
```

Seems like this should return an error.  Since your recursive calls below are checking for `dict` instances, this would only return `None` if the parent `nested_dict` were not a `dict`.  Making this an error would make debugging easier.

---

[e2c9a51](https://github.com/NOAA-EMC/global-workflow/pull/2387/commits/e2c9a512444ff805fd31252b244b42a0a6cc4200) update will ensure a `KeyError` is thrown when the key is not found.
Additionally, a `TypeError` is caught if the incoming data is not a `dict`.